onnxscript/rewriter/pattern.py

```diff
@@ -1374,8 +1381,14 @@ def try_rewrite(
             check_match_result = self._condition_function(context, **match.bindings)
             if not check_match_result:
                 # If check function was provided, but it failed, return the reason for failure to the tracer.
-                if not isinstance(check_match_result, bool):
-                    match.fail(check_match_result.reason)
+                if isinstance(check_match_result, MatchResult):
+                    if check_match_result._failure_nodes_and_values:
```

Comment:
nit: this if-then-else condition is not required? I think the then branch should work correctly in both cases, right?